node/header_exchange.go

```diff
@@ -0,0 +1,241 @@
+	if err := hExService.headerStore.Init(ctx, initial); err != nil {
+		return err
+	}
+	if err := hExService.syncer.Start(hExService.ctx); err != nil {
+		return err
+	}
+	hExService.syncerStarted = true
+	return nil
+}
+
+func (hExService *HeaderExchangeService) tryInitHeaderStoreAndStartSyncer(ctx context.Context, trustedHeader *types.Header) {
+	if trustedHeader != nil {
+		if err := hExService.initHeaderStoreAndStartSyncer(ctx, trustedHeader); err != nil {
+			hExService.logger.Error("failed to initialize the headerstore and start syncer", "error", err)
+		}
+	} else {
+		signedHeader := <-hExService.syncedHeadersCh
+		if signedHeader.Header.Height() == hExService.genesis.InitialHeight {
+			if err := hExService.initHeaderStoreAndStartSyncer(ctx, &signedHeader.Header); err != nil {
+				hExService.logger.Error("failed to initialize the headerstore and start syncer", "error", err)
+			}
+		}
+	}
+}
+
+func (hExService *HeaderExchangeService) writeToHeaderStoreAndBroadcast(ctx context.Context, signedHeader *types.SignedHeader) {
+	// Init the header store if first block, else append to store
+	if err := hExService.initOrAppendHeaderStore(ctx, &signedHeader.Header); err != nil {
+		hExService.logger.Error("failed to write block header to header store", "error", err)
+	}
+
+	// Broadcast for subscribers
+	if err := hExService.sub.Broadcast(ctx, &signedHeader.Header); err != nil {
+		hExService.logger.Error("failed to broadcast block header", "error", err)
+	}
+}
+
+// OnStart is a part of Service interface.
+func (hExService *HeaderExchangeService) Start() error {
+	var err error
+	// have to do the initializations here to utilize the p2p node which is created on start
+	ps := hExService.p2p.PubSub()
+	hExService.sub = goheaderp2p.NewSubscriber[*types.Header](ps, pubsub.DefaultMsgIdFn)
+	if err = hExService.sub.Start(hExService.ctx); err != nil {
+		return fmt.Errorf("error while starting subscriber: %w", err)
+	}
+	if _, err := hExService.sub.Subscribe(); err != nil {
+		return fmt.Errorf("error while subscribing: %w", err)
+	}
+
+	if err = hExService.headerStore.Start(hExService.ctx); err != nil {
+		return fmt.Errorf("error while starting header store: %w", err)
+	}
+
+	_, _, network := hExService.p2p.Info()
+	if hExService.p2pServer, err = newP2PServer(hExService.p2p.Host(), hExService.headerStore, network); err != nil {
+		return err
+	}
+	if err = hExService.p2pServer.Start(hExService.ctx); err != nil {
+		return fmt.Errorf("error while starting p2p server: %w", err)
+	}
+
+	peerIDs := hExService.p2p.PeerIDs()
+	if hExService.ex, err = newP2PExchange(hExService.p2p.Host(), peerIDs, network, hExService.p2p.ConnectionGater()); err != nil {
+		return err
+	}
+	if err = hExService.ex.Start(hExService.ctx); err != nil {
+		return fmt.Errorf("error while starting exchange: %w", err)
+	}
+
+	// for single aggregator configuration, syncer is not needed
+	// TODO (ganesh): design syncer flow for multiple aggregator scenario
```

Comment:
[non-blocking] @gupadhyaya can you create an issue for this, explaining what needs to be done for this ?

Reply:
https://github.com/rollkit/rollkit/issues/749